src/main/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/GCNVPostProcessor.java

```diff
@@ -0,0 +1,197 @@
+
+        /* INFO header lines*/
+        result.addMetaDataLine(new VCFInfoHeaderLine(VCFConstants.END_KEY, 1,
+                VCFHeaderLineType.Integer, "End coordinate of this variant"));
+        outputWriter.writeHeader(result);
+    }
+
+    /**
+     * Write variant context fields given all posterior records from a single gCNV output chunk
+     *
+     * @param copyNumberPosteriorLocatableRecordsChunk records from a gCNV output chunk
+     * @param variantPrefix variant id prefix
+     */
+    void writeChunkedVariantContext(final List<CopyNumberPosteriorLocatableRecord> copyNumberPosteriorLocatableRecordsChunk,
+                                    final String variantPrefix) {
+        for (CopyNumberPosteriorLocatableRecord posteriorRecord: copyNumberPosteriorLocatableRecordsChunk) {
+            final VariantContext variantContext = composeVariantContext(posteriorRecord, variantPrefix);
+            outputWriter.add(variantContext);
+            //check that analyzed posterior records are ordered according to the provided sequence dictionary
+            if (lastPosteriorRecord != null) {
+                if (IntervalUtils.compareLocatables(lastPosteriorRecord, posteriorRecord, samSequenceDictionary) > 0) {
+                    throw new UserException.BadInput("Intervals in the chunk directories are not" +
+                            " ordered according to the sequence dictionary");
+                }
+            }
+            lastPosteriorRecord = posteriorRecord;
+        }
+    }
+
+    /**
+     * Compose a variant context given a posterior record for an interval
+     *
+     * @param copyNumberPosteriorLocatableRecord a posterior record to genotype
+     * @param variantPrefix a variant prefix
+     * @return composed variant context
+     */
+    private VariantContext composeVariantContext(final CopyNumberPosteriorLocatableRecord copyNumberPosteriorLocatableRecord,
+                                                 final String variantPrefix) {
+        final VariantContextBuilder variantContextBuilder = new VariantContextBuilder();
+        variantContextBuilder.alleles(alleles);
+        variantContextBuilder.chr(copyNumberPosteriorLocatableRecord.getContig());
+        variantContextBuilder.start(copyNumberPosteriorLocatableRecord.getStart());
+        variantContextBuilder.stop(copyNumberPosteriorLocatableRecord.getEnd());
+        variantContextBuilder.id(String.format(variantPrefix + "_%s_%d_%d",
+                copyNumberPosteriorLocatableRecord.getContig(),
+                copyNumberPosteriorLocatableRecord.getStart(),
+                copyNumberPosteriorLocatableRecord.getEnd()));
+        final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(sampleName);
+        final List<Integer> copyNumberPLVector = getCopyNumberPLVector(copyNumberPosteriorLocatableRecord, integerCopyNumberStateCollection);
```

Comment:
Can the reviewer check carefully the math here?:)

Reply:
i'll leave this to you, @mbabadi, i've got to stop here!

Reply:
Done